lib/sendgrid/helpers/mail/personalization.rb

```diff
@@ -43,6 +43,7 @@ def add_header(header)
 
     def add_substitution(substitution)
       substitution = substitution.to_json
+      return if substitution['substitution'].nil?
```

Comment:
I understand the current error being thrown is not ideal, but this effectively swallows the issue. What's the use case where it's okay to attempt to add a substitution that's _invalid_?

Reply:
For my use case, I have a `SendgridMailer` utility that handles the sending of all emails from my other mailers.

Calling the `SendgridMailer` class might look something like this:

```ruby
SendgridMailer.send_email(
  template: 'friendly-template-name',
  to: user.email,
  options: {
    from: 'hello@email.com',
    from_name: 'Hello Mailer',
    subject: 'Here is an email'
  },
  substitutions: { 
    '-FIRST_NAME-' => user.first_name, 
    '-LAST_NAME-' => user.last_name,
    '-ADDRESS-' => user.address
  }
)
```

And my `send_email` looks something like this:

```ruby
def self.send_email(template:, to:, options:, substitutions: {})
  mail = SendGrid::Mail.new
  mail.from = SendGrid::Email.new(email: options[:from], name: options[:from_name])
  mail.subject = options[:subject]
  mail.template_id = TEMPLATE_MAP[template] # maps friendly name to UUID

  personalization = SendGrid::Personalization.new
  personalization.add_to(SendGrid::Email.new(email: to))
  substitutions.each do |key, value|
    personalization.add_substitution(SendGrid::Substitution.new(key: key, value: value))
  end
  mail.add_personalization(personalization)

  sendgrid = SendGrid::API.new(api_key: SENDGRID_API_KEY)
  sendgrid.client.mail._('send').post(request_body: mail.to_json)
end
```

The iteration of substitutions will break if the value for one is nil.

The crux of the failure is that `Substitution#to_json` [deletes the `substitution` attribute](https://github.com/sendgrid/sendgrid-ruby/blob/c1aa3082254cc3e88b3ee9489b0165aed6c3b782/lib/sendgrid/helpers/mail/substitution.rb#L16-L20) if the value is nil, but the `Personalization#add_substitution` method expects it to be present.

Reply:
I understand why it would fail, but the alternative in this case would be something like the substitution for `-LAST_NAME-` not getting replaced and thus showing up as _-LAST_NAME-_ in the sent mail. It's still an error in my view, and a more proper error should be raised.

Reply:
I definitely get that and it makes sense, but it feels weird that it's allowed to fail like it does. Maybe instead of allowing it to reach `Personalization#add_substitution` before failing, `Substitution.new` could raise an error if the value is nil?

Reply:
That _could_ be a more breaking change as it's perfectly reasonable to instantiate a new `Substitution`, then populate it (and maybe update it), then add it to the `Personalization` (not sure why you'd choose to do it this way, but you could).

I think it feels more natural to error when attempting to add a malformed substitution to a personalization.

Reply:
Understood; I’ll defer to your expertise. I’ll close this PR.